### PR TITLE
[ng] repatriate dart sample code + make fragments locally

### LINKED ISSUE #206: [ng] migrate e2e tests from ng.io
This includes migration of the:
- example code
- e2e tests
- Relevant portion of e2e testing infrastructure.

### LINKED ISSUE #207: [ng] port or find equivalent to ng.io shredder (file fragment extractor)

### LINKED ISSUE #222: [ng] build code fragments locally vs copying from ng.io

### PR DESCRIPTION
**Pull request by chalin**
We're reusing the dgeni-based shredder (fragment maker) for now.

Fixes #206
Fixes #207 
Fixes #222

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -31,3 +32,17 @@ node_modules
 # Temp content copied over from angular.io
 /src/resources
 /src/angular/_fragments
+
+# /public/**
```
Comment (kwalrath) [inline .gitignore]: delete this?

File: .gitignore [hunk 2]
```diff
@@ -31,3 +32,17 @@ node_modules
...
+
+# Examples
+
+.packages
+.pub
+build
+pubspec.lock
+a2docs.css
+# !/public/docs/_examples/
```
Comment (kwalrath) [inline .gitignore]: Why have these commented-out lines in .gitignore? To track changes in the ng.io source?

File: gulp/examples.js [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+  const EXAMPLES_TESTING_PATH = path.join(EXAMPLES_PATH, 'testing/ts');
+  const BOILERPLATE_PATH = path.join(EXAMPLES_PATH, '_boilerplate');
+  const TOOLS_PATH = config.TOOLS_PATH;
+  const STYLES_SOURCE_PATH = path.join(TOOLS_PATH, 'styles-builder/less');
+
+  const lang='dart';
+
+  var _exampleBoilerplateFiles = [
+    'a2docs.css',
+    // 'package.json',
```
Comment (kwalrath) [inline gulp/examples.js]: Are these commented out so it's easier to compare this file to the ng.io equivalent, and keep it up to date as that one changes?

**Comment by chalin** [comment]
As discussed offline, this is a milestone commit. Further work is coming and the commented out code will be either reinstated or removed.

**Pull request merged**